task.go

```diff
@@ -187,6 +192,21 @@ func (e *Executor) RunTask(ctx context.Context, call taskfile.Call) error {
 		return err
 	}
 
+	e.generateLock.Lock()
+	for _, g := range t.Generates {
```

Comment:
Keep in mind that `t.Generates` can contains a glob instead of a file path.

In an ideal implementation, you'd expand this glob and lock each of the returned files.

Reply:
This initial pass was intended to work with globs since the user is declaring what the task is going to modify.  The problem with checking for any found files for the glob is that any files being generated may not exist yet, so nothing will be returned and either nothing will lock or everything will lock.

That being said, there are definitely some holes.  If Task A generates `output/file.data` and Task B generates `output/**/*.data`, then we likely have a problem.  If Task A generates `output/**/*.jpg` and Task B generates `output/*.jpg` there could be some overlap as well.

One improvement might be to double down on the locks; basically use the method already in the PR, and then *also* lock on glob matches.  But that's inconsistent depending on whether the files exist already or not, and to me that's a large problem and makes that approach unappealing.

Another approach would be to try to be smart about the generated field matches.  I could imagine some basic logic being able to tell that there's overlap between `output/**/*.data` and `output/file.data`, for example.  This would give us consistency regardless of the state of generation and also give the user a lot more safety without having to worry about their config being a certain way.